src/Sylius/Component/User/Security/Generator/UniquePinGenerator.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Component\User\Security\Generator;
+
+use Sylius\Component\User\Security\Checker\UniquenessCheckerInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Łukasz Chruściel <lukasz.chrusciel@lakion.com>
+ * @author Michał Marcinkowski <michal.marcinkowski@lakion.com>
+ * @author Jan Góralski <jan.goralski@lakion.com>
+ */
+final class UniquePinGenerator implements GeneratorInterface
+{
+    /**
+     * @var UniquenessCheckerInterface
+     */
+    protected $uniquenessChecker;
+
+    /**
+     * @var int
+     */
+    protected $tokenLength;
+
+    /**
+     * @param UniquenessCheckerInterface $uniquenessChecker
+     * @param int $tokenLength
+     *
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(UniquenessCheckerInterface $uniquenessChecker, $tokenLength)
+    {
+        Assert::integer(
+            $tokenLength,
+            'The value of pin length has to be an integer.'
+        );
+        Assert::range(
+            $tokenLength,
+            1, 9,
+            'The value of pin length has to be in range between 1 to 9.'
+        );
+
+        $this->tokenLength = $tokenLength;
+        $this->uniquenessChecker = $uniquenessChecker;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function generate()
+    {
+        do {
+            $token = $this->randomize();
+        } while (!$this->uniquenessChecker->isUnique($token));
+
+        return $token;
+    }
+
+    /**
+     * @return string
+     */
+    private function randomize()
+    {
+        $min = pow(10, $this->tokenLength - 1);
+        $max = pow(10, $this->tokenLength) - 1;
+
+        return (string) mt_rand($min, $max);
```

Comment:
This generator will not generate codes with leading zeros, can we make it something like this instead?
```php
strpad((string) mt_rand(0, pow(10, $this->tokenLength) - 1), $this->tokenLength, '0', STR_PAD_LEFT);
```